python/lsst/sims/catUtils/mixins/sncat.py

```diff
@@ -533,7 +601,79 @@ def get_snparams(self):
             self.column_by_name('cosmologicalDistanceModulus')
 
         vals = self.SNparamDistFromHost(hostz, hostid, hostmu)
-
         return (vals[:, 0], vals[:, 1], vals[:, 2], vals[:, 3])
 
 
+    def get_snid(self):
+        # Not necessarily unique if the same galaxy hosts two SN
+        # Use refIdCol to access the relevant id column of the dbobj
+        # Should revert to galTileID for galaxyTiled catalogDBObj and
+        # id for galaxyObj catalogDBObj
+        # (email from Scott)
+        return self.column_by_name(self.refIdCol)
+
+class FrozenSNCat(SNFunctionality,  InstanceCatalog, CosmologyMixin, SNUniverse):
+
+    """
+    `lsst.sims.catalogs.measures.instance.InstanceCatalog` class with SN
+    characterized by the  following attributes
+
+    Attributes
+    ----------
+    column_outputs :
+    suppressHighzSN :
+    maxTimeSNVisible :
+    maxz :
+    variables :
+    override_formats :
+    cannot_be_null :
+    mjdobs :
+    badvalues position :
+    3-tuple of floats (ra, dec, redshift), velocity : 3 tuple of floats
+        velocity wrt host galaxy in Km/s, the supernova model (eg. SALT2)
+    and parameters of the supernova model that predict the SED.
+    """
+
+    surveyStartDate = 59580. # For Kraken_1042 / Minion_1016
+
+
+
+    @compound('snra', 'sndec', 'z', 'vra', 'vdec', 'vr')
+    def get_angularCoordinates(self):
+        '''
+        Obtain the coordinates and velocity of the SN from the host galaxy
+
+        Returns
+        -------
+        `np.ndarray` of coordinara, dec, z, vra, vdec, and vr
+
+        '''
+        snra, sndec, snz = self.column_by_name('raJ2000'),\
+            self.column_by_name('decJ2000'),\
+            self.column_by_name('Tredshift')
+        snvra = np.zeros(self.numobjs)
+        snvdec = np.zeros(self.numobjs)
```

Comment:
One of the things that makes getters (as implemented in InstanceCatalog) so useful is that you can always sub-class something like SNIaCatalog and define a getter for snvra and snvdec.  I'm not sure how that will work now that we have a default getter for them (in fact, I think it won't work; InstanceCatalog will complain if there are two getters for the same column).  If snvra and snvdec are not defined here, then users are free to write their own method to calculate them.  That method can still access snra and sndec from column_by_name().

If you don't want to deal with this now, that is fine.  I think we should look at this question in the future, though.